File: Frontend/src/Types/planta.ts
export interface Planta {
  id: string;
  nombre: string;
  region: string;
  idComp: string;
  estado: boolean;
  fecha_log: string;
}

export interface PlantaResponse {
  data: Planta[];
  message: string;
  success: boolean;
}

export interface PlantaCreateRequest {
  id: string;
  nombre: string;
  idComp: string;
  region: string;
  estado: boolean;
  fecha_log: string;
}

export interface PlantaUpdateRequest extends PlantaCreateRequest {
  id: string;
}

File: Backend/maintenace-service/src/Entity/Planta.cs
namespace Entity
{
    public class Planta
    {
        public string? Id { get; set; }              // Identificador único de la planta
        public string? Nombre { get; set; }          // Nombre de la planta
        public string? Region { get; set; }          // Región donde se encuentra la planta
        public string IdComp { get; set; }          // FK de la compañía asociada
        public bool Estado { get; set; }            // Estado (activo/inactivo)
        public DateTime? Fecha_log { get; set; }     // Fecha de creación o última modificación
    }
}